### PR TITLE
feat: added generateStaticParams to use ISR with the dynamic route for deals

### PR DESCRIPTION
**Pull request by klae32**
## Description

This PR uses the getStaticProps using ISR to allow the data from the db to be cahched as well as make sure to improve performance overall
## Issue

Closes or Relates to: #194 
## PR Requirements

- [x] I have carefully read through and understand the [Deals-for-Devs Contributing Guide](https://github.com/Learn-Build-Teach/deals-for-devs/blob/dev/.github/contributing.md)
- [x] The title of this PR follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) format
- [x] The `Description` gives a good representation of the changes made
- [x] If this PR addresses an open Issue, it is linked in the `Issue` section

### CODE REVIEW

File: src/app/deals/[id]/page.tsx [hunk 1]
```diff
@@ -10,6 +10,18 @@ type Props = {
   searchParams: { [key: string]: string | string[] | undefined }
 }
 
+export async function getStaticProps() {
```
Comment (jamesqquick) [inline src/app/deals/[id]/page.tsx:13]: This solution should use the `generateStaticParams` function. You can find the docs here, https://nextjs.org/docs/app/api-reference/functions/generate-static-params.

**Review by jamesqquick — CHANGES_REQUESTED** [review]

File: src/app/deals/[id]/page.tsx (hunk 1 above)
Comment (klae32) [reply src/app/deals/[id]/page.tsx]: @jamesqquick I was able to use the proper function and got it working properly as well as being statically generated. See photo below
Screenshot 2024-07-13 at 8 32 30 AM

**Comment by klae32** [comment]
@jamesqquick Did you get a chance to take a look at these changes?

File: src/app/deals/[id]/page.tsx [hunk 2]
```diff
@@ -10,6 +10,14 @@ type Props = {
   searchParams: { [key: string]: string | string[] | undefined }
 }
 
+export async function generateStaticParams() {
+  const deals = await getApprovedDeals()
+
+  return deals.map((deal) => ({
+    deal: deal.xata_id,
```
[the remark this answers is not in the document]
Comment (klae32) [reply src/app/deals/[id]/page.tsx:17]: Made the change here.

File: src/app/deals/[id]/page.tsx [hunk 3]
```diff
@@ -10,6 +10,14 @@ type Props = {
   searchParams: { [key: string]: string | string[] | undefined }
```
[the remark this answers is not in the document]
Comment (klae32) [reply src/app/deals/[id]/page.tsx:10]: Removed. Resolving this one.

File: src/app/deals/[id]/page.tsx [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import { notFound } from 'next/navigation'
-import { getDealById } from '@/lib/queries'
+import { getApprovedDeals, getDealById } from '@/lib/queries'
 import DealPreview from '@/components/DealPreview'
 import { Metadata, ResolvingMetadata } from 'next'
```
[the remark this answers is not in the document]
Comment (klae32) [reply src/app/deals/[id]/page.tsx:5]: Great call out @jamesqquick, This is updated, resolving now.

**Review by jamesqquick — APPROVED** [review]

**Pull request merged**